docs/proposals/2017-04-27 Text Formatting.md

```diff
@@ -0,0 +1,239 @@
+
+This is useful for:
+
+* _Importing Content_ - Many legacy CMSes, including ones in use at Washington Post, store content as HTML, and thus having a direct pathway for sections of HTML content into ANS is expedient;
+* _WYSIWYG Web Editor_ - Virtually all web browsers support the "contenteditable" attribute to some degree, so using HTML for some of the formatting allows web based editors (such as Ellipsis) to take advantage of that fact;
+* _HTML Templates_ - Simple HTML templates can copy the text value as-is with no additional parsing or rendering.
+
+Any successor solution should seek to be as useful or more to the above end-users.
+
+
+## Goals of this Proposal
+
+According to this proposal, the ideal text formatting solution should also be feasible to use in:
+
+* _Rainbow_ - Existing rainbow products already consume JSON formatted content;
+* _iOS, Android, Apple News, and other non-HTML environments_ - Apple in particular is enthusiastic about its design-oriented Apple News format, which has features without direct HTML analogues;
+* _Elasticsearch_ - It is a critical piece of the Content API with unique limitations on how data is stored that should be addressed in the schema when possible (HTML is not the worst, not the best format here).
+
+It should also:
+
+* Maintain existing HTML options (when practical) as a convenience options for current & similar ANS users;
+* Limit text formatting to a reasonable set of options (somewhere in the range of "superscript yes, kerning no");
+* Address existing limitations in representing element-element relationships that are not strictly sequential, and element-document relationships.
+
+
+# Proposal
+
+## Part I. Structured Text Object.
+
+### `attributed_text` object describes structured text content
+
+Attributed text object will have the following structure. Note that proposed structure is compatible with preexisting `text` story element (as of 0.5.8).
+
+* a required `text` property with plain text content of type `string`,
+* an optional `attributes` property with array of `attribute` objects describing formatting to be applied to the text,
+* an optional `selections` property with array of `selection` objects describing text sub-strings and formatting,
+* an optional `content` property with HTML formatted text content.
+
+Example:
+```
+  {
+    "_id": "AAAAAAAAAAAAAAAAAAAAAAAAAA",
+    "type": "text",
+    "text": "The text attribute should contain plain text content only.",
+    "selections": [ ... ]
+    "content": "The <b>text</b> attribute should contain <i>plain text</i> content only."
+  }
+```
+
+### `selection`s describe formatting for specific segments of a text string
+
+Each `selection` object defines a part of a text string and what formatting attributes should be applied to it. An idea borrowed from Apple and Twitter.
```

Comment:
If I have text that I need to show up in an editor app but I don't want displayed in a published story, such as a note between a copy editor and a reporter, would there be a way to denote that with this schema?